File: py-writer-workspace/evals/eval-1-follow-patterns/project/app/services/user_service.py
from app.models.user import User
from app.repositories.base import UserRepository
from app.exceptions import NotFoundError, ConflictError


class UserService:
    def __init__(self, repo: UserRepository) -> None:
        self._repo = repo

    async def get_by_id(self, user_id: str) -> User:
        user = await self._repo.find_by_id(user_id)
        if user is None:
            raise NotFoundError("User", user_id)
        return user

    async def create(self, email: str, name: str) -> User:
        existing = await self._repo.find_by_email(email)
        if existing is not None:
            raise ConflictError(f"Email already in use: {email}")
        return await self._repo.create(email=email, name=name)

    async def list_all(self) -> list[User]:
        return await self._repo.find_all()

File: tdd-workspace/evals/eval-1-feature-with-infra/project/src/services/userService.ts
import { db } from '../db';

export class UserService {
  async getById(id: string) {
    const user = await db.users.findById(id);
    if (!user) throw new Error('User not found');
    return user;
  }

  async create(email: string, name: string) {
    const existing = await db.users.findByEmail(email);
    if (existing) throw new Error('Email already in use');
    return db.users.create({ email, name });
  }
}
